### PR TITLE
[oneMKL] Updated oneMKL API for Sparse Conjugate Gradient 

### PR DESCRIPTION
**Pull request by andreyfe1**
# Existing Sample Changes
## Description

Updated oneMKL API for the Sparse Conjugate Gradient sample

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [ ] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

### CODE REVIEW

File: Libraries/oneMKL/sparse_conjugate_gradient/sparse_cg.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ void run_sparse_cg_example(const sycl::device &dev)
     try {
         mkl::sparse::init_matrix_handle(&handle);
 
-        mkl::sparse::set_csr_data(handle, nrows, nrows, mkl::index_base::zero,
+        mkl::sparse::set_csr_data(main_queue, handle, nrows, nrows, mkl::index_base::zero,
                                           ia_buffer, ja_buffer, a_buffer);
```
Comment (spencerpatty) [inline Libraries/oneMKL/sparse_conjugate_gradient/sparse_cg.cpp:142]: these are good, but we shold also make one more change to remove deprecated TRSV usage as well -- 

Line 227 - 235 should be changed to the following (added 1.0 before handle for alpha parameter)

```
            {
                mkl::sparse::trsv(main_queue, mkl::uplo::lower,
                                          mkl::transpose::nontrans,
                                          mkl::diag::nonunit, 1.0, handle, r_buffer, t_buffer);
                diagonal_mv<fp, intType>(main_queue, nrows, d_buffer, t_buffer);
                mkl::sparse::trsv(main_queue, mkl::uplo::upper,
                                          mkl::transpose::nontrans,
                                          mkl::diag::nonunit, 1.0, handle, t_buffer, w_buffer);
            }
```

**Review by spencerpatty — APPROVED** [review]
approved as is, and recommendation for one more change.

File: Libraries/oneMKL/sparse_conjugate_gradient/sparse_cg.cpp (hunk 1 above)
Comment (andreyfe1) [reply Libraries/oneMKL/sparse_conjugate_gradient/sparse_cg.cpp:142]: Thanks. Fixed

**Comment by andreyfe1** [comment]
@jimmytwei , could you please merge the changes?

**Pull request merged**